Document link order of SPIRV-Tool libraries.

I did a pull of the tree today and rebuilt my project, but there are now linker errors that I can't seem to resolve. I'm linking `libSPIRV-Tools.a opt/libSPIRV-Tools-opt.a link/libSPIRV-Tools-link.a` but still, symbols are missing:

```
/home/lritter/devel/duangle/scopes/SPIRV-Tools/build/source/opt/libSPIRV-Tools-opt.a(pass_manager.cpp.o): In function `spvtools::opt::PassManager::Run(spvtools::ir::IRContext*)':
pass_manager.cpp:(.text+0x34): undefined reference to `spvutils::PrintTimerDescription(std::ostream*, bool)'
pass_manager.cpp:(.text+0x102): undefined reference to `vtable for spvutils::Timer'
pass_manager.cpp:(.text+0x125): undefined reference to `spvutils::Timer::Start()'
pass_manager.cpp:(.text+0x176): undefined reference to `spvutils::Timer::Report(char const*)'
pass_manager.cpp:(.text+0x4f8): undefined reference to `spvutils::Timer::Report(char const*)'
/home/lritter/devel/duangle/scopes/SPIRV-Tools/build/source/opt/libSPIRV-Tools-opt.a(pass_manager.cpp.o): In function `spvutils::ScopedTimer<spvutils::Timer>::~ScopedTimer()':
pass_manager.cpp:(.text._ZN8spvutils11ScopedTimerINS_5TimerEED2Ev[_ZN8spvutils11ScopedTimerINS_5TimerEED5Ev]+0x25): undefined reference to `spvutils::Timer::Report(char const*)'
/home/lritter/devel/duangle/scopes/SPIRV-Tools/build/source/opt/libSPIRV-Tools-opt.a(pass_manager.cpp.o): In function `spvutils::ScopedTimer<spvutils::Timer>::~ScopedTimer()':
pass_manager.cpp:(.text._ZN8spvutils11ScopedTimerINS_5TimerEED0Ev[_ZN8spvutils11ScopedTimerINS_5TimerEED5Ev]+0x25): undefined reference to `spvutils::Timer::Report(char const*)'
/home/lritter/devel/duangle/scopes/SPIRV-Tools/build/source/opt/libSPIRV-Tools-opt.a(vector_dce.cpp.o): In function `spvtools::opt::VectorDCE::AddItemToWorkListIfNeeded(spvtools::opt::VectorDCE::WorkListItem, std::unordered_map<unsigned int, spvtools::utils::BitVector, std::hash<unsigned int>, std::equal_to<unsigned int>, std::allocator<std::pair<unsigned int const, spvtools::utils::BitVector> > >*, std::vector<spvtools::opt::VectorDCE::WorkListItem, std::allocator<spvtools::opt::VectorDCE::WorkListItem> >*)':
vector_dce.cpp:(.text+0x1711): undefined reference to `spvtools::utils::BitVector::Or(spvtools::utils::BitVector const&)'
```

I tried the stable branch, but the same error exists there.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -300,7 +300,7 @@ if (NOT "${SPIRV_SKIP_TESTS}")
            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
 endif()
 
-set(SPIRV_LIBRARIES "-lSPIRV-Tools -lSPIRV-Tools-link -lSPIRV-Tools-opt")
+set(SPIRV_LIBRARIES "-lSPIRV-Tools-opt -lSPIRV-Tools -lSPIRV-Tools-link")
 set(SPIRV_SHARED_LIBRARIES "-lSPIRV-Tools-shared")
 
 # Build pkg-config file

